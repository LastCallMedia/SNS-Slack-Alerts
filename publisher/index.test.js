
const AlertFactory = require('./');

describe('AlertFactory', function() {
    const realDateNow = () => Date.now();
    beforeEach(function() {
        global.Date.now = jest.fn(() => 1530518207007);
    })
    afterEach(function() {
        global.Date.now = realDateNow;
    })

    it('Should create a formatted Slack message', function() {
        const factory = new AlertFactory('foo', 'bar');
        const message = factory.create('Subject', 'Message');
        message.addLink('Test', 'https://google.com');
        expect(message.formatSlack()).toMatchSnapshot();
    });

    it('Should create a formatted SNS message', function() {
        const factory = new AlertFactory('foo', 'bar');
        const message = factory.create('Subject', 'Message');
        expect(message.formatSNS()).toMatchSnapshot();
    });

    // @todo: Test Lambda alert creation.
});