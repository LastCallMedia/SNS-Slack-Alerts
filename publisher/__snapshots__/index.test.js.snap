// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertFactory Should create a formatted SNS message 1`] = `
Object {
  "Message": "{\\"default\\":\\"Message\\",\\"lambda\\":\\"{\\\\\\"attachments\\\\\\":[{\\\\\\"text\\\\\\":\\\\\\"Message\\\\\\",\\\\\\"fallback\\\\\\":\\\\\\"Message\\\\\\",\\\\\\"actions\\\\\\":[],\\\\\\"footer\\\\\\":\\\\\\"bar\\\\\\",\\\\\\"ts\\\\\\":1530518207.007}]}\\"}",
  "MessageStructure": "json",
  "Subject": "Subject",
}
`;

exports[`AlertFactory Should create a formatted Slack message 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "text": "Test",
          "type": "button",
          "url": "https://google.com",
        },
      ],
      "fallback": "Message",
      "footer": "bar",
      "text": "Message",
      "ts": 1530518207.007,
    },
  ],
}
`;
